services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/HankeServiceITests.kt

```diff
@@ -854,6 +854,40 @@ class HankeServiceITests : DatabaseTest() {
         assertThat(alue.geometriat).isNotNull
     }
 
+    @Test
+    fun `updateHanke new hankealue name updates name and keeps data intact`() {
+        val createdHanke = hankeService.createHanke(getATestHanke())
+        val hankealue = createdHanke.alueet[0]
+        assertNull(hankealue.nimi)
+        val modifiedHanke =
+            createdHanke.copy().apply {
+                // manually set mutable collections due to a shallow copy.
+                this.omistajat = createdHanke.omistajat
+                this.arvioijat = createdHanke.arvioijat
+                this.toteuttajat = createdHanke.toteuttajat
+                this.tyomaaTyyppi = createdHanke.tyomaaTyyppi
+                this.alueet = mutableListOf(hankealue.copy(nimi = "Changed Name"))
+            }
+
+        val updateHankeResult = hankeService.updateHanke(modifiedHanke)
+
+        assertThat(updateHankeResult)
+            .usingRecursiveComparison()
+            .ignoringFields("modifiedAt", "modifiedBy", "version", "alueet")
+            .isEqualTo(createdHanke)
+        assertEquals(createdHanke.alueet.size, updateHankeResult.alueet.size)
```

Comment:
Mixing `assertThat` and `assertEquals` in the same test is a bit confusing, since they have different order for expected and actual values.

The expected value here should be always 1, since you reset the alueet in the update call. I.e. here `this.alueet = mutableListOf(hankealue.copy(nimi = "Changed Name"))`.

Reply:
Assertion is now assertThat and expected value is 1.

Reply:
Writing this as `assertThat(createdHanke.alueet).hasSize(1)` gives a more descriptive error message if the assert fails.

There's another `assertEquals` for the nimi.